Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/FilterScreen.kt

```diff
@@ -0,0 +1,276 @@
+                            } else {
+                                ContentAlpha.disabled
+                            }
+                            CompositionLocalProvider(LocalContentAlpha provides alpha) {
+                                Text(
+                                    text = stringResource(id = R.string.reset),
+                                    style = MaterialTheme.typography.body2
+                                )
+                            }
+                        }
+                    },
+                    backgroundColor = JetsnackTheme.colors.uiBackground
+                )
+            }
+        ) {
+            Column(
+                Modifier
+                    .fillMaxSize()
+                    .verticalScroll(rememberScrollState())
+                    .padding(horizontal = 24.dp, vertical = 16.dp),
+            ) {
+                SortFiltersSection(
+                    sortState = sortState,
+                    onFilterChange = { filterName ->
+                        sortChanged = filterName != SnackRepo.getSortDefault()
+                        sortState = filterName
+                    }
+                )
+                FilterChipSection(
+                    title = stringResource(id = R.string.price),
+                    filters = priceFilters
+                )
+                FilterChipSection(
+                    title = stringResource(id = R.string.category),
+                    filters = categoryFilters
+                )
+
+                MaxCalories(
+                    sliderPosition = maxCalories,
+                    onValueChanged = { newValue ->
+                        maxCalories = newValue
+                    }
+                )
+                FilterChipSection(
+                    title = stringResource(id = R.string.lifestyle),
+                    filters = lifeStyleFilters
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun FilterChipSection(title: String, filters: List<Filter>) {
+    FilterTitle(text = title)
+    FlowRow(
+        mainAxisAlignment = FlowMainAxisAlignment.Center,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(top = 12.dp, bottom = 16.dp)
+            .padding(horizontal = 4.dp)
+    ) {
+        filters.forEach { filter ->
+            FilterChip(
+                filter = filter,
+                modifier = Modifier.padding(end = 4.dp, bottom = 8.dp)
+            )
+        }
+    }
+}
+
+@Composable
+fun SortFiltersSection(sortState: String, onFilterChange: (String) -> Unit) {
+    FilterTitle(text = stringResource(id = R.string.sort))
+    Column(Modifier.padding(bottom = 24.dp)) {
+        SortFilters(
+            sortState = sortState,
+            onChanged = onFilterChange
+        )
+    }
+}
+
+@Composable
+fun SortFilters(
+    sortFilters: List<Filter> = SnackRepo.getSortFilters(),
+    sortState: String,
+    onChanged: (String) -> Unit
```

Comment:
What about making this `(Filter) -> Unit`?

Reply:
Done